components/director/internal/domain/oauth20/service.go

```diff
@@ -72,6 +90,22 @@ func (s *service) CreateClientCredentials(ctx context.Context, objectType model.
 	return credentialData, nil
 }
 
+func (s *service) UpdateClientScopes(ctx context.Context, clientID string, objectType model.SystemAuthReferenceObjectType) error {
+	scopes, err := s.GetClientCredentialScopes(objectType)
+	if err != nil {
+		if !model.IsIntegrationSystemNoTenantFlow(err, objectType) {
```

Comment:
There are few `GetClientCredentialScopes` calls and the returned error is handled differently. Here and in `CreateClientCredentials` we are looking for `IsIntegrationSystemNoTenantFlow`.(in sanitize directive we do not check for this kind of error) I could not get why this check is needed at all.